src/core/modes-text.ts

```diff
@@ -293,7 +293,7 @@ function parse(
             let atoms: Atom[];
             [atoms, tokens] = options.parse('text', tokens, options);
             result = [...result, ...atoms];
-        } else if (token.length === 1) {
+        } else if ([...token].length === 1) {
```

Comment:
This is for correctly handling characters like 💩 that take up two UTF-16 characters. Is that fine?
(See https://mathiasbynens.be/notes/javascript-unicode )

Reply:
OK, it took me a while to figure out (a) what your fix was doing and (b) what was the underlying problem, but I got it now. It's a good idea, but it's only a partial fix, and it has some performance implications (converting the string to an array of codepoints with the spread operator for each token).

It's only a partial fix because of a subtlety of Unicode encoding where some characters can actually be more than two codepoints, and wouldn't be correctly handled (or are not guaranteed to be handled) by the Javascript spread operator. For example the David Bowie 👨🏻‍🎤 emoji is made up of U+1F468 (man) + U+200D (zero-width-joiner) + U+1F3A4 (microphone). There is already code in Mathlive that handles this (see `graphemesplitter`), so that this combination will be considered a single token.

It is indeed incorrect to check for the length here. However, the better fix is simply to move the code that's in the `if (token.length ===1)` branch into the `else {}` branch (and just remove the error call). If the token is *not* one of the special ones, we should just handle it, without checking the length of the token first.

Reply:
Oh, thank you! I learned something new about Unicode, that's great.